GSA/digitalgov.gov
# Generate webp responsive images

## Summary
Update responsive image scripts to convert `jpg|png` to `webp`. Support for `webp` is about [96% global users](https://caniuse.com/webp).

> [!NOTE]
> Current support for this will only touch new image uploads going forward 

## Preview

[Link to Preview]()

## Solution

Reducing the number of variant sizes to 2 covers most of our image needs and preferring `webp` format fits with current best practices for image serving.

- Replace `jpg|png` variants to only generate `webp` images.
- Generate only `800` for mobile/tablet and `1200` for desktop/widescreen
- Update template to display `800` and `1200` depending on viewport


## How To Test

1. Run local instance and upload test image with `npx gulp upload`
2. Should generate `800` & `1200` webp versions and original image
3. Should generate `yaml` image data file
4. 3 images should be uploaded to S3
5. Images are not corrupted on S3, can be viewed from S3 link

## What To Test
1. Visit [it-warning-banners-how-gsa-is-working-to-stop-unnecessarily-frightening-users/](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-add-responsive-webp/2024/08/07/it-warning-banners-how-gsa-is-working-to-stop-unnecessarily-frightening-users/)
6. Check network tab is serving `webp` and not `png`.
7. Regression test [evaluating-your-agency-accessibility-just-got-a-whole-lot-easier-with-gsa-openacr-editor](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-add-responsive-webp/2024/07/08/evaluating-your-agency-accessibility-just-got-a-whole-lot-easier-with-gsa-openacr-editor/) still serves `png` image
